docs/migration/v12.md

```diff
@@ -0,0 +1,329 @@
+# Platform v12 migration guide: Vite & React 18
+
+Good news! Vite and React 18 in the app platform are ready to use!
+
+We’re very excited for these updates — both will significantly modernize the App Platform, and Vite will be a big upgrade from Create React App, which we used under the hood to build and serve apps previously. It will greatly improve the developer experience, and with greater control over the configuration, it will open up some powerful new possibilities for the platform.
+
+Here are some tips about what to expect, and how to easily upgrade to the latest version of `@dhis2/cli-app-scripts` to take advantage of Vite and React 18.
+
+### Notable changes
+
+These are some things that you’ll see right away:
+
+-   It’s fast! Starting up an app is nearly instant, and building an app is about 3-4 times faster than before
+-   Plugins are handled better:
+    -   Start-up of both the app and plugin is nearly instant
+    -   The app and plugin are run on the same port
+    -   Support for a plugin without an app is improved
+    -   Code is shared between entrypoints, which makes bundles smaller
+    -   HMR for code changes in the plugin will be as fast as in the app
+-   There are some new TypeScript features:
+    -   Bootstrapping an app with a TypeScript template is supported: `d2-app-scripts init --typescript my-app`
+    -   Vite has native support for TypeScript
+    -   Although not in the App Platform package, with the latest `@dhis2/cli-style`, TypeScript type checking is performed when running `d2-style lint`
+    -   With `@dhis2/ui` version 9, the UI library is now typed as well
+-   There’s a small suite of tools available in the CLI when running an app in dev mode:
+    -   With the dev server running, press `h + enter` to see the options
+    -   Options include exposing the server on LAN, opening the app in the browser, cleanly quitting the server, and restarting the dev server (which can be helpful if you’re modifying libraries in node_modules — see more below)
+-   The build output includes a summary to inspect chunks
+
+### Future changes
+
+With Vite, the door is open for some big future improvements:
```

Comment:
Just a note about this section, this will need adjusted once these features are either deployed or dismissed. I think it is fine to have it here but we must not forget. 

If there are already JIRA tickets/epics about these features, we should link it to this file as a todo.